flux-build-scripts/update-flux-versions.js

```diff
@@ -70,13 +70,27 @@ function addVersion(versionObj, influxdbVer, fluxVer) {
   versionObj[influxdbVer] = fluxVer
 }
 
+// Compare semantic minor versions
+function isSemanticNewer(semverX, semverY) {
+  xMajor = parseInt(semverX.split(".")[0])
+  xMinor = parseInt(semverX.split(".")[1])
+  yMajor = parseInt(semverY.split(".")[0])
+  yMinor = parseInt(semverY.split(".")[1])
+
+  if (xMajor > yMajor) {return true}
+  else if (xMajor == yMajor) {
+    if (xMinor >= yMinor) {return true}
+    else {return false}
+  } else {return false}
+}
+
 // Load product data from the product data file
 const productData = yaml.load(fs.readFileSync(path.resolve(__dirname,'../data/products.yml'), 'utf8'))
 
-// Update InfluxDB version arrays by removing 'v' from each version and filtering
-// out InfluxDB versions that don't have a Flux dependency in their go.mod
-const ossVersionArr = productData.influxdb.minor_versions.map((element, index) => {return element.replace('v', '')}).filter(element => parseFloat(element) >= 1.7).reverse();
-const enterpriseVersionArr = productData.enterprise_influxdb.minor_versions.map((element, index) => {return element.replace('v', '')}).filter(element => parseFloat(element) >= 1.9).reverse();
+// Update InfluxDB version arrays by removing 'v' from each version
+// Filter out InfluxDB versions that don't have a Flux dependency in their go.mod
+const ossVersionArr = productData.influxdb.minor_versions.map((element) => {return element.replace('v', '')}).filter(version => isSemanticNewer(version, '1.7'));
+const enterpriseVersionArr = productData.enterprise_influxdb.minor_versions.map((element) => {return element.replace('v', '')}).filter(version => isSemanticNewer(version, '1.9'));
```

Comment:
Magic number

Reply:
1.9 is the version of Enterprise that started getting regular updates with Flux and Flux was added to the `go.mod`. The other versions of enterprise have fixed versions.